claudette/core.py

```diff
@@ -332,11 +333,20 @@ def _repr_markdown_(self:Client):
 | Server tool use | {self.use.server_tool_use.web_search_requests:,} | {server_tool_cost:.6f} |
 | **Total** | **{self.use.total:,}** | **${self.cost:.6f}** |"""
 
+# %% ../00_core.ipynb
+@dataclass
+class ToolResult:
+    result_type: str
+    data: Any
+    def __str__(self): return str(self.data)
+    def __repr__(self): return f"ToolResult(result_type='{self.result_type}', data={self.data!r})"
+
 # %% ../00_core.ipynb
 def mk_funcres(fc, ns):
     "Given tool use block `fc`, get tool result, and create a tool_result response."
     res = call_func(fc.name, fc.input, ns=ns, raise_on_err=False)
-    return dict(type="tool_result", tool_use_id=fc.id, content=str(res))
+    if isinstance(res, ToolResult): return dict(type="tool_result", tool_use_id=fc.id, content=str(res.data))
+    else: return dict(type="tool_result", tool_use_id=fc.id, content=str(res))
 
```

Comment:
Could get rid of the dupe code here using something like:

```suggestion
    if isinstance(res, ToolResult): res = res.data
    return dict(type="tool_result", tool_use_id=fc.id, content=str(res))
```